wire-library/wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -214,6 +214,8 @@
         for (Field field : fieldsAndOneOfFields) {
           String suggestion = collidingNames.contains(field.getName())
             || schema.getType(field.getQualifiedName()) != null
+            || (field.getName().equals(field.getType().getSimpleName())
+              && !field.getType().isScalar())
```

Comment:
Let's match what the KotlinGenerator does:
```suggestion
            || field.getName().equals(field.getType().getSimpleName())
            || schema.getType(field.getQualifiedName()) != null
```